tests/validation/tests/v3_api/test_zero_downtime.py

```diff
@@ -0,0 +1,352 @@
+from .common import *
+import pytest
+import math
+import time
+from .test_rke_cluster_provisioning import create_custom_host_from_nodes
+
+HOST_NAME = os.environ.get('RANCHER_HOST_NAME', "testZeroDT")
+DRAIN = os.environ.get('RANCHER_DRAIN_INPUT', "False")
+MAX_UNAVAILABLE_WORKER = os.environ.get("RANCHER_MAX_UNAVAILABLE_WORKER", "10%")
+MAX_UNAVAILABLE_CONTROLPLANE = os.environ.get("RANCHER_MAX_UNAVAILABLE_CONTROLPLANE", "1")
+NODE_COUNT_CLUSTER = os.environ.get("RANCHER_NODE_COUNT_CLUSTER", 3)
+host = "test" + str(random_int(10000, 99999)) + ".com"
+path = "/name.html"
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None,
+             "nodes": []}
+backup_info = {"backupname": None, "backup_id": None, "workload": None,
+               "backupfilename": None, "etcdbackupdata": None}
+K8S_VERSION_PREUPGRADE = os.environ.get("RANCHER_K8S_VERSION_PREUPGRADE", "v1.15.12-rancher2-2")
+K8S_VERSION_UPGRADE = os.environ.get("RANCHER_K8S_VERSION_UPGRADE", "v1.16.10-rancher2-1")
+
+
+def test_zero_downtime():
+    client, cluster, workload, ingress, rke_config = get_and_validate_cluster()
+    # Update Cluster to k8 version + upgrade strategy maxUnavailable worker
+    rke_updated_config = rke_config.copy()
+    rke_updated_config["kubernetesVersion"] = K8S_VERSION_UPGRADE
+    rke_updated_config["upgradeStrategy"] = {
+        'drain': True,
+        'maxUnavailableWorker': MAX_UNAVAILABLE_WORKER,
+        'maxUnavailableControlplane': MAX_UNAVAILABLE_CONTROLPLANE,
+        'type': '/v3/schemas/nodeUpgradeStrategy'}
+    cluster = client.update(cluster,
+                            name=cluster.name,
+                            rancherKubernetesEngineConfig=rke_updated_config)
+    completed_nodes = validate_nodes_after_upgrade(cluster, workload)
+    nodes = client.list_node(clusterId=cluster.id).data
+    assert len(completed_nodes) == len(nodes), "Not all Nodes Upgraded"
+    node_ver = K8S_VERSION_UPGRADE.split("-")[0]
+    for node in nodes:
```

Comment:
maybe I am missing something, why didn't we include this in some validate func as well?